### PR TITLE
Prometheus: Fix prom links in mixed mode

### LINKED ISSUE #21129: Bug in Prometheus links with --mixed-- data source
## What happened

1. Have 2 Prometheus data sources.
2. Create a panel with `--mixed--` data source.
3. Add 2 queries with different Prometheus data sources.

When viewing the panel in edit mode, the **Link to Graph in Prometheus** of every data source has the domain name of the first data source.

For example:

Query A uses data source `http://prom-1:9090`  
Query B uses data source `http://prom-2:9090`

Link A will be `http://prom-1:9090/graph?g0.expr=query-A`  
Link B will be `http://prom-1:9090/graph?g0.expr=query-B`

Notice only the domain part is wrong. The query expression is correct.

## What you expected to happen

Query A uses data source `http://prom-1:9090`  
Query B uses data source `http://prom-2:9090`

Link A will be `http://prom-1:9090/graph?g0.expr=query-A`  
Link B will be `http://prom-2:9090/graph?g0.expr=query-B`

## How to reproduce it (as minimally and precisely as possible)

I attached a tarball that reproduces the problem in a Docker container.

[prometheus-mixed-datasource-bug.tar.gz](https://github.com/grafana/grafana/files/3970280/prometheus-mixed-datasource-bug.tar.gz)

```
tar tzf prometheus-mixed-datasource-bug.tar.gz 
prometheus-mixed-datasource-bug/
prometheus-mixed-datasource-bug/start.sh
prometheus-mixed-datasource-bug/datasources.yml
prometheus-mixed-datasource-bug/dashboards.yml
prometheus-mixed-datasource-bug/dashboards/
prometheus-mixed-datasource-bug/dashboards/a.json
```

Running `start.sh` runs a Grafana container and provisions two data sources and 1 dashboard.

Open http://localhost:3000/d/ashboard/prometheus-links-in-mixed-data-source?orgId=1&fullscreen&edit&panelId=2 and look at the URLs.

Compare the two Prometheus Graph URLs:

```
http://prom-1:9090/graph?g0.expr=up&g0.range_input=21600s&g0.end_input=2019-12-16%2021%3A41&g0.step_input=30&g0.tab=0
http://prom-1:9090/graph?g0.expr=node_cpu_seconds_total&g0.range_input=21600s&g0.end_input=2019-12-16%2021%3A41&g0.step_input=30&g0.tab=0
```

## Environment
- Grafana version: 6.5.2
- Data source type & version: Prometheus
- OS Grafana is installed on: Docker
- User OS & Browser: Fedora, Firefox
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by zoltanbedi**
**Which issue(s) this PR fixes**:

Fixes #21129

### CODE REVIEW

**Review by aocenas — APPROVED** [review]
Looks good, seems to fail on some flaky backend test

File: public/app/plugins/datasource/prometheus/components/PromLink.tsx [hunk 1]
```diff
@@ -36,23 +35,17 @@ export default class PromLink extends Component<Props, State> {
...
-    const datasourceName = target.datasource;
-    const datasource: PrometheusDatasource = datasourceName
-      ? (((await getDatasourceSrv().get(datasourceName)) as any) as PrometheusDatasource)
-      : (this.props.datasource as PrometheusDatasource);
-
     const range = request.range;
-    const start = datasource.getPrometheusTime(range.from, false);
-    const end = datasource.getPrometheusTime(range.to, true);
+    const start = this.props.datasource.getPrometheusTime(range.from, false);
+    const end = this.props.datasource.getPrometheusTime(range.to, true);
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/prometheus/components/PromLink.tsx:40]: Any reason why we are not destructuring this.props so we could to just `datasource.getPrometheusTime(range.from, false);`? We are already doing ` const { query } = this.props;`.

**Review by ivanahuckova — APPROVED** [review]
I have added one nitpick comment/question. Other than that LGTM! 🕺 Awesome that you have added test coverage for this as well.

File: public/app/plugins/datasource/prometheus/components/PromLink.tsx (hunk 1 above)
Comment (zoltanbedi) [reply public/app/plugins/datasource/prometheus/components/PromLink.tsx:40]: Good finding! Let me destruct datasource then.

**Pull request merged**